<h1>Анализ вакансий для Junior Data Analyst на портале LinkedIn<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Введение" data-toc-modified-id="Введение-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Введение</a></span></li><li><span><a href="#Цель-исследования" data-toc-modified-id="Цель-исследования-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Цель исследования</a></span></li><li><span><a href="#Задачи-исследования" data-toc-modified-id="Задачи-исследования-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Задачи исследования</a></span></li><li><span><a href="#Источник-данных" data-toc-modified-id="Источник-данных-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Источник данных</a></span></li><li><span><a href="#Структура-датасета" data-toc-modified-id="Структура-датасета-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Структура датасета</a></span></li><li><span><a href="#Используемые-библиотеки" data-toc-modified-id="Используемые-библиотеки-6"><span class="toc-item-num">6&nbsp;&nbsp;</span>Используемые библиотеки</a></span></li><li><span><a href="#Знакомство-с-датасетом" data-toc-modified-id="Знакомство-с-датасетом-7"><span class="toc-item-num">7&nbsp;&nbsp;</span>Знакомство с датасетом</a></span></li><li><span><a href="#Работа-с-пропусками" data-toc-modified-id="Работа-с-пропусками-8"><span class="toc-item-num">8&nbsp;&nbsp;</span>Работа с пропусками</a></span></li><li><span><a href="#Изменение-типов-данных" data-toc-modified-id="Изменение-типов-данных-9"><span class="toc-item-num">9&nbsp;&nbsp;</span>Изменение типов данных</a></span></li><li><span><a href="#Перевод-текстовых-данных-на-английский-язык" data-toc-modified-id="Перевод-текстовых-данных-на-английский-язык-10"><span class="toc-item-num">10&nbsp;&nbsp;</span>Перевод текстовых данных на английский язык</a></span></li><li><span><a href="#Проверка-на-дубликаты" data-toc-modified-id="Проверка-на-дубликаты-11"><span class="toc-item-num">11&nbsp;&nbsp;</span>Проверка на дубликаты</a></span></li><li><span><a href="#Исследовательский-анализ-данных" data-toc-modified-id="Исследовательский-анализ-данных-12"><span class="toc-item-num">12&nbsp;&nbsp;</span>Исследовательский анализ данных</a></span><ul class="toc-item"><li><span><a href="#Выборка-из-аналитиков-согласно-ТЗ" data-toc-modified-id="Выборка-из-аналитиков-согласно-ТЗ-12.1"><span class="toc-item-num">12.1&nbsp;&nbsp;</span>Выборка из аналитиков согласно ТЗ</a></span></li><li><span><a href="#Выстаскиваем-hard-skills" data-toc-modified-id="Выстаскиваем-hard-skills-12.2"><span class="toc-item-num">12.2&nbsp;&nbsp;</span>Выстаскиваем hard skills</a></span></li><li><span><a href="#Подготовка-данных-для-дашборда:-удаление-лишних-столбцов" data-toc-modified-id="Подготовка-данных-для-дашборда:-удаление-лишних-столбцов-12.3"><span class="toc-item-num">12.3&nbsp;&nbsp;</span>Подготовка данных для дашборда: удаление лишних столбцов</a></span></li><li><span><a href="#Приложение-1---Ссылка-на-Дашборд" data-toc-modified-id="Приложение-1---Ссылка-на-Дашборд-12.4"><span class="toc-item-num">12.4&nbsp;&nbsp;</span>Приложение 1 - Ссылка на Дашборд</a></span></li></ul></li></ul></div>

## Введение

LinkedIn является одним из ведущих онлайн-сообществ, объединяющим профессионалов со всего мира. Портал предоставляет уникальную возможность исследовать требования, навыки, и опыт, необходимые для успешной карьеры в области анализа данных. В рамках исследования будет проведен анализ различных аспектов вакансий для Junior Data Analyst на LinkedIn. Будут рассмотрены популярные сферы, в которых востребованы специалисты по анализу данных. Особое внимание уделено требованиям работодателей к кандидатам (hard skills), необходимые для успешной работы в данной сфере. Кроме того, будет осуществлен анализ типов занятости и локаций работы для Junior Data Analyst в странах и городах Европы. Результаты исследования, оформленные в виде дашборда, позволят профессиональным аналитикам, студентам и выпускникам сферы анализа данных принять более информированное решение о своей карьере и выборе работы за рубежом. Это исследование принесет практическую пользу и поможет определиться с дальнейшими шагами в карьерном развитии в сфере анализа данных.

## Цель исследования

Визуализировать информацию о рынке вакансий для Junior Data Analyst в Европе.

## Задачи исследования

Подготовка данных:
1. выгрузить датасет из LinkedIn и провести его обзор;
2. провести обработку данных: заполнить пропуски, проверить данные на дублирование, изменить тип данных при необходимости;
3. перевести данные на один язык (английский);
4. исключить из перечня вакансий тимлидов, сеньоров и миддлов;
5. вытащить из описания вакансии hard skills, которые работодатель требует от соискателя;
6. удалить лишние столбцы датафрейма, которые не пригодятся для визуализации информации.

Визуализация данных:
Построить интерактивный дашборд в любой из BI систем (с учетом возможности расшарить этот дашборд). По требованию от заказчика дашборд должен содержать:
1. `фильтры` — по стране и по типу занятости;
2. количество вакансий (абсолютные значения) – `индикатор`;
3. количество вакансий по странам (относительные значения) — `stack bar chart`;
4. тип занятости — `pie chart`;
5. список нанимающих компаний с указанием количества вакансий, отсортированный в порядке убывания - `heat map`;
6. ТОП 10 сфер деятельности компаний, которые нанимают аналитиков — `barchart`;
7. размер компаний и количество вакансий — `pie chart`;
8. hard skills — `barchart`;
9. зависимость количества кандидатов на вакансию от даты публикации объявления — `линейный график`;
10. количество вакансий по городам - `map`.

## Источник данных

Датасет Аналитик + bi аналитик (джун) - https://disk.yandex.ru/d/Rlo2KdJRve6fkw

## Структура датасета

1. `job_title` - название вакансии;
2. `area` - полное название места работы;
3. `city` - город места работы;
4. `country` - страна места работы;
5. `workplace` - тип занятости;
6. `company_name` - название компании-работодателя;
7. `industry` - сфера деятельности компании-работодателя;
8. `company_size` - размер компании-работодателя;
9. `job_description` - описание вакансии;
10. `posted_date` - дата публикации вакансии;
11. `applicants` - количество откликов на вакансию.

Дополнительно в ходе исследования будут созданы два новых столбца:
1. `right_posted_date` - производный от столбца `posted_date` с датой в формате ггг.мм.дд
2. `hard skills` - производный от столбца `job_description`, где собраны ключевые hard skills.


## Используемые библиотеки

In [1]:
# Импорт вспомогательных библиотек
import pandas as pd # для работы с таблицей 
import dateparser # для работы с датами
from googletrans import Translator # для перевода текста
import re # регуляторные выражения для вычленения hard skills

## Знакомство с датасетом

In [2]:
# Выгрузка датасета и сохранение его в переменной df:
df = pd.read_csv('C:\\Users\\MSI\\Desktop\\Project\\masterskaya_LinkedIn_2023_05.csv')

In [3]:
# Знакомство с датасетом:
df.head()

Unnamed: 0                        job_title  \
0           0                     Data Analyst   
1           1         Data Analyst - Logistics   
2           2         Data Analyst - Logistics   
3           3  Data Analyst (Space & Planning)   
4           4                     Data Analyst   

                                            area          city  \
0              ['Basel', 'Basel', 'Switzerland']         Basel   
1      ['Coventry', 'England', 'United Kingdom']      Coventry   
2      ['Coventry', 'England', 'United Kingdom']      Coventry   
3  ['South Molton', 'England', 'United Kingdom']  South Molton   
4            ['Lugano', 'Ticino', 'Switzerland']        Lugano   

          country workplace                              company_name  \
0     Switzerland   On-site  PharmiWeb.Jobs: Global Life Science Jobs   
1  United Kingdom   On-site                      Resolute Recruitment   
2  United Kingdom   On-site                      Resolute Recruitment   
3  United Kingdom   On-site                       Mole Valley Farmers   
4     Switzerland   On-site                                   FORFIRM   

                industry company_size  \
0  Staffing & Recruiting        11-50   
1                    NaN          NaN   
2                    NaN          NaN   
3                    NaN          NaN   
4                    NaN          NaN   

                                     job_description  posted_date  applicants  
0  About the job\n        \n\n \n                ...   1 week ago        47.0  
1                                      About the job   1 week ago         NaN  
2  About the job\n        \n\n \nData Analyst - L...   1 week ago         NaN  
3  About the job\n        \n\n \nSalary: To be di...   1 week ago         NaN  
4  About the job\n        \n\n \nFORFIRM is provi...  2 weeks ago         NaN

In [4]:
# Знакомство с датасетом
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 998 entries, 0 to 997
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Unnamed: 0       998 non-null    int64  
 1   job_title        998 non-null    object 
 2   area             998 non-null    object 
 3   city             998 non-null    object 
 4   country          998 non-null    object 
 5   workplace        930 non-null    object 
 6   company_name     996 non-null    object 
 7   industry         964 non-null    object 
 8   company_size     963 non-null    object 
 9   job_description  998 non-null    object 
 10  posted_date      998 non-null    object 
 11  applicants       838 non-null    float64
dtypes: float64(1), int64(1), object(10)
memory usage: 93.7+ KB


Данные успешно загружены и первая информация по ним успешно получена. Датасет состоит из 12 столбцов и 998 строк. Названия столбцов приведены к змеиному регистру, что значительно упрощает исследование. У каждого столбца есть две проблемы: неверный тип данных и наличие пропущенных значений (`workplace`, `company_name`, `industry`, `company_size`, `applicants`).

## Работа с пропусками

In [5]:
# Знакомство со значениями столбца workplace:
df['workplace'].value_counts()

workplace
Hybrid     486
On-site    343
Remote     101
Name: count, dtype: int64

In [6]:
# Замена пропусков на значение no datas:
df['workplace'] = df['workplace'].fillna('No datas')
df['workplace'].value_counts()

workplace
Hybrid      486
On-site     343
Remote      101
No datas     68
Name: count, dtype: int64

Пропущенные значения столбца `workplace` мы классифицируем как `MNAR`. Так как нам неизвестна природа этих пропусков, то самым оптимальным решением будет поставить заглушку в виде `No datas`.

In [7]:
# Удаление пропущенных строк в столбцах company_name,
# industry, company_size
df = df.dropna(subset=['company_name', 'industry','company_size'])

Данный шаг был выполнен, поскольку столбцы нужны для дальнейших подсчетов. Если подобные вакансии с пропущенными значениями оставить, то они все равно не будут учитываться в визуализации, а само исследование выйдет некорректным, если, допустим, в одной строке столбец с названием компании будет заполнен, а с индустрией - нет.

Следующим шагом будет проанализировать столбец `applicants`. Из условия мы знаем, что здесь сохранено количество откликов на вакансии. Разумно предположить гипотезу о том, что пропуски связаны с тем, что на вакансию не откликались. Но прежде чем заполнить пропуски значением 0, убедимся, что такого значения нет в данных:

In [8]:
df.query('applicants == 0')

Empty DataFrame
Columns: [Unnamed: 0, job_title, area, city, country, workplace, company_name, industry, company_size, job_description, posted_date, applicants]
Index: []

In [9]:
# Замена пропущенных значений в столбце applicants на 0
df['applicants'] = df['applicants'].fillna(0)

В столбце aplicants сохранено количество откликов на вакансию. Пропущенные значения в строках говорят о том, что на эту вакансию соискатели не откликались, поэтому будет разумным заменить их на 0.

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 963 entries, 0 to 997
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Unnamed: 0       963 non-null    int64  
 1   job_title        963 non-null    object 
 2   area             963 non-null    object 
 3   city             963 non-null    object 
 4   country          963 non-null    object 
 5   workplace        963 non-null    object 
 6   company_name     963 non-null    object 
 7   industry         963 non-null    object 
 8   company_size     963 non-null    object 
 9   job_description  963 non-null    object 
 10  posted_date      963 non-null    object 
 11  applicants       963 non-null    float64
dtypes: float64(1), int64(1), object(10)
memory usage: 97.8+ KB


В ходе первичной предобработки данных было удалено 35 строк, что составлен 3,5% от всего датафрейма, а значит на основную картину никак не появлияет.

## Изменение типов данных

In [11]:
# Для столбцов, содержащих потенциально текстовые данные,
# деляем преобразование в текст, ставим нижний регистр
# удаляем лишние пробелы:
df['job_title'] = (df['job_title']
                 .astype('string')
                 .str.lower()
                 .str.strip()
                )

df['area'] = (df['area']
                 .astype('string')
                 .str.lower()
                 .str.strip()
                )
df['city'] = (df['city']
                 .astype('string')
                 .str.lower()
                 .str.strip()
                )

In [12]:
# Для столбцов, содержащих потенциально текстовые данные,
# деляем преобразование в текст, ставим нижний регистр
# удаляем лишние пробелы:
df['country'] = (df['country']
                 .astype('string')
                 .str.lower()
                 .str.strip()
                )

df['workplace'] = (df['workplace']
                 .astype('string')
                 .str.lower()
                 .str.strip()
                )
df['company_name'] = (df['company_name']
                 .astype('string')
                 .str.lower()
                 .str.strip()
                )

In [13]:
# Для столбцов, содержащих потенциально текстовые данные,
# деляем преобразование в текст, ставим нижний регистр
# удаляем лишние пробелы:
df['industry'] = (df['industry']
                 .astype('string')
                 .str.lower()
                 .str.strip()
                )

df['company_size'] = (df['company_size']
                 .astype('string')
                 .str.lower()
                 .str.strip()
                )
df['job_description'] = (df['job_description']
                 .astype('string')
                 .str.lower()
                 .str.strip()
                )

In [14]:
# С датой поступаем так же, как и с текстом
# Для дальнейшего преобразования
df['posted_date'] = (df['posted_date']
                 .astype('string')
                 .str.lower()
                 .str.strip()
                )
# Количество откликов - числа целые:
df['applicants'] = (df['applicants']
                 .astype('int')
                )

Приведение текста к нижнему регистру и удаление лишних пробелов было сделано для упрощения процесса нахождения потенциальных дубликатов.

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 963 entries, 0 to 997
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Unnamed: 0       963 non-null    int64 
 1   job_title        963 non-null    string
 2   area             963 non-null    string
 3   city             963 non-null    string
 4   country          963 non-null    string
 5   workplace        963 non-null    string
 6   company_name     963 non-null    string
 7   industry         963 non-null    string
 8   company_size     963 non-null    string
 9   job_description  963 non-null    string
 10  posted_date      963 non-null    string
 11  applicants       963 non-null    int32 
dtypes: int32(1), int64(1), string(10)
memory usage: 94.0 KB


In [16]:
# Создаем новый столбец для преобразования текста в дату
# средствами библиотеки dataparser:
df['right_posted_date'] = (df['posted_date'].apply(
    lambda x: dateparser.parse(x))
                          )
# Для применения команды ко всему столбцу пишем
# простую лямбда-функцию

In [17]:
# Приводим новый столбец в формат даты
df['right_posted_date'] = pd.to_datetime(df['right_posted_date']).dt.date

In [18]:
# Смотрим результат преобразований
df.head()

Unnamed: 0                      job_title  \
0            0                   data analyst   
23          23                data analyst ii   
25          25  business intelligence analyst   
26          26             data analyst (f/h)   
27          27                     bi analyst   

                                              area        city      country  \
0                ['basel', 'basel', 'switzerland']       basel  switzerland   
23          ['dublin', 'county dublin', 'ireland']      dublin      ireland   
25             ['cracow', 'małopolskie', 'poland']      cracow       poland   
26  ['dijon', 'bourgogne-franche-comté', 'france']       dijon       france   
27               ['alsónémedi', 'pest', 'hungary']  alsónémedi      hungary   

   workplace                              company_name  \
0    on-site  pharmiweb.jobs: global life science jobs   
23   on-site  pharmiweb.jobs: global life science jobs   
25    hybrid                                   getinge   
26    hybrid                                  neos-sdi   
27    hybrid                                       gls   

                             industry company_size  \
0               staffing & recruiting        11-50   
23              staffing & recruiting        11-50   
25                     medical device      10,001+   
26  information technology & services       51-200   
27   transportation/trucking/railroad      10,001+   

                                      job_description  posted_date  \
0   about the job
        

 
                what...   1 week ago   
23  about the job
        

 
                titl...   1 week ago   
25  about the job
        

 
at getinge we have t...  3 weeks ago   
26  about the job
        

 
l’équipe data de dij...   1 week ago   
27  about the job
        

 
a gls dinamikusan nö...   1 week ago   

    applicants right_posted_date  
0           47        2024-01-09  
23          30        2024-01-09  
25          70        2023-12-26  
26         141        2024-01-09  
27          36        2024-01-09

Все данные в столбцах преобразованы к нужному типу данных.

## Перевод текстовых данных на английский язык

Для перевода данных к единому языку воспользуемся средствами библиотеки `googletrans`:

In [19]:
# Смотрим список вакансий
df['job_title'].value_counts()

job_title
data analyst                                              83
online data analyst                                       38
data analyst - milano                                     28
data analyst (m/w/d)                                      25
datový analytik                                           19
                                                          ..
l.t.e. s.p.a. - ingegnere di processo                      1
operational specialist                                     1
consultant senior microsoft bi - power bi expert h/f       1
application support specialist - business intelligence     1
analista de datos bi                                       1
Name: count, Length: 532, dtype: Int64

Видно, что данные в датасете мультиязычные (по мимо английского - чешский, польский, немецкий, итальянский и т.д.), от того есть потребность перевести информацию, содержащуюся в таблице, к единому языку. В данном исследовании будет разумно перевести все на английский язык, так как данные с использованием кириллицы могут отображаться некорректно.

In [20]:
# Средствами библиотеки traclator переводим
# информацию в таблице на английский язык
translator = Translator()
df['job_title']= df['job_title'].apply(lambda x: translator.translate(x, dest='en').text)
# Для применения перевода ко всему столбцу job_title
# используется простая лямбда-функция

In [21]:
# Проверка
df['job_title'].value_counts()

job_title
data analyst                      117
online data analyst                38
data analyst - milano              28
data analyst (m/w/d)               25
data analyst:in                    19
                                 ... 
product manager                     1
rehabilitation project manager      1
course coordination employee        1
seller for the Skaraborg area       1
bi data analyst                     1
Name: count, Length: 520, dtype: int64

In [22]:
# Аналогично поступаем и со столбцом job_description,
# которые тоже является мультиязычным
df ['job_description']= df['job_description'].apply(lambda x: translator.translate(x, dest='en').text)

В данном шаге исследования вся информация в таблице была приведена к единому языку. Это упростит задачу по отсеиванию не-джунов и по вычленению hard skills.

## Проверка на дубликаты

In [23]:
df.duplicated().sum()

0

Дубликаты отсутствуют.

## Исследовательский анализ данных

### Выборка из аналитиков согласно ТЗ

In [24]:
# Смотрим список вакансий:
list(df['job_title'].unique())

['data analyst',
 'data analyst ii',
 'business intelligence analyst',
 'data analyst (f/h)',
 'bi analyst',
 'data analyst associate',
 'product data analyst',
 'graduate data analyst',
 'digital data analyst',
 'business intelligence analyst junior',
 'data analyst h/f',
 'data analyst (m/f/d)',
 'analytics consultant',
 'data analyst f/h',
 'bi consultant',
 'statistical data analyst',
 'digital data analyst (h/f)',
 'data analyst - €60,- per hour - amsterdam based',
 'data & analytics senior analyst',
 'junior data analyst – aerospace',
 'data analyst - operations',
 'data analyst (h/f)',
 'business intelligence analyst (m/w/d)',
 'senior data analyst',
 'junior data analyst',
 'web data analyst f/h',
 'business analyst_power bi',
 'junior data analyst power bi - 6 month contract',
 'data analyst - milano',
 'data analyst (engineer)',
 'data analyst h/f _ cdd',
 'data analyst - data analyst',
 'bi analyst, power bi champion, cee based in warsaw',
 'data analyst tableau',
 'sql/data

Не все в списке вакансий являются аналитиками - одних можно отнести к вспомогательному персоналу, в других вакансиях ищут человека на позиции middle, senior. Согласно цели исследования нас интересуют только позиции джунов. Так как во многих вакансиях нет приставки Junior, будем считать, что они тоже являются джуновскими.

In [25]:
# Сохраняем в список ключевые слова,
keywords = ['data analyst', 'science', 'bi-analyst', 
            'bi analyst', 'bi', 'business analyst', 
            'marketing analyst']
# по которым будем отбирать нужные нам вакансии
# средствами str.contains()
filtered_data = df[df['job_title'].str.contains('|'.join(keywords), case=False, na=False)]

In [26]:
# Список слов, которые необходимо исключить
exclude_words = ['lead', 'team-lead', 
                 'team lead', 'senior', 
                 'middle', 'master']
# Создаем фильтр, который исключает строки, содержащие указанные слова
filter = ~filtered_data['job_title'].str.contains('|'.join(exclude_words), case=False)
# Применяем фильтр к датафрейму
filtered_data = filtered_data[filter]

In [27]:
# Полученный результат:
filtered_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 615 entries, 0 to 997
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Unnamed: 0         615 non-null    int64 
 1   job_title          615 non-null    object
 2   area               615 non-null    string
 3   city               615 non-null    string
 4   country            615 non-null    string
 5   workplace          615 non-null    string
 6   company_name       615 non-null    string
 7   industry           615 non-null    string
 8   company_size       615 non-null    string
 9   job_description    615 non-null    object
 10  posted_date        615 non-null    string
 11  applicants         615 non-null    int32 
 12  right_posted_date  615 non-null    object
dtypes: int32(1), int64(1), object(3), string(8)
memory usage: 64.9+ KB


Таким образом, были отобраны джун-вакансии и результаты сохранены в переменную `filtered_data`.

### Выстаскиваем hard skills

In [28]:
# Список хард скиллов
hard_skills = ["python", "database", "r", "sql", "data mining", "excel", "react", "statistic",
              'power bi', 'bi analysis', 'bi-analysis', 'tableau', 'oracle', 'etl',
              'mysql', 'postgresql', 'mongodb', 'pandas', 'numpy', 'scipy', 'matplotlib',
              'cloud', 'google metric', 'business intelligence', 'big data',
              'lean analytics', 'six sigma', 'machine learning', 'artificial intelligence', 'ai',
              'math', 'economy', 'marketing', 'gdpr', 'pr', 'public relations', 'cohort analysis',
              'cohort', 'powerpoint', 'data visualization', 'wrangling', 'data wrangling',
              'data manipulation', 'analytics', 'sas', 'spss', 'qlikview', 'nosql', 'presentation',
              'bpmn', 'aris', 'visio', 'jira', 'agile', 'kanban', 'trello', 'uml', 'erp', 'parsing',
              'eda', 'exploratory data analysis', 'seaborn', 'plotly', 'plotly express']

# Функция для вычленения хард скиллов из описания вакансии
def extract_hard_skills(description):
    extracted_skills = []
    for skill in hard_skills:
        if re.search(r'\b' + re.escape(skill) + r'\b', description, re.IGNORECASE):
            extracted_skills.append(skill)
    return extracted_skills

# Вычленение хард скиллов и сохранение их в отдельном столбце
filtered_data['hard_skills'] = filtered_data['job_description'].apply(lambda x: extract_hard_skills(x))

# Вывод таблицы
filtered_data.head()

Unnamed: 0           job_title  \
0            0        data analyst   
23          23     data analyst ii   
26          26  data analyst (f/h)   
27          27          bi analyst   
29          29        data analyst   

                                              area        city  \
0                ['basel', 'basel', 'switzerland']       basel   
23          ['dublin', 'county dublin', 'ireland']      dublin   
26  ['dijon', 'bourgogne-franche-comté', 'france']       dijon   
27               ['alsónémedi', 'pest', 'hungary']  alsónémedi   
29     ['manchester', 'england', 'united kingdom']  manchester   

           country workplace                              company_name  \
0      switzerland   on-site  pharmiweb.jobs: global life science jobs   
23         ireland   on-site  pharmiweb.jobs: global life science jobs   
26          france    hybrid                                  neos-sdi   
27         hungary    hybrid                                       gls   
29  united kingdom   on-site                                axon moore   

                             industry company_size  \
0               staffing & recruiting        11-50   
23              staffing & recruiting        11-50   
26  information technology & services       51-200   
27   transportation/trucking/railroad      10,001+   
29              staffing & recruiting       51-200   

                                      job_description  posted_date  \
0   about the job\n        \n\n \n                ...   1 week ago   
23  about the job\n        \n\n \n                ...   1 week ago   
26  about the job\n        \n\n \nthe Dijon data t...   1 week ago   
27  about the job\n        \n\n \ngls' dynamically...   1 week ago   
29  about the job\n        \n \n\n\n\n      this j...  3 weeks ago   

    applicants right_posted_date  \
0           47        2024-01-09   
23          30        2024-01-09   
26         141        2024-01-09   
27          36        2024-01-09   
29          81        2023-12-26   

                                          hard_skills  
0   [database, sql, data mining, excel, analytics,...  
23                                 [machine learning]  
26  [sql, power bi, business intelligence, data ma...  
27          [database, sql, power bi, tableau, agile]  
29                    [python, sql, excel, analytics]

Выборка именно таких hard skills обусловена всесторонним анализом европейского рынка труда и требований работодателей к умениями и навыкам на позицию джуниор. Список может расширятся по запросу Заказчика.

In [29]:
# Не забываем обновлять индексы:
filtered_data = filtered_data.reset_index(drop=True)
filtered_data.head()

Unnamed: 0           job_title  \
0           0        data analyst   
1          23     data analyst ii   
2          26  data analyst (f/h)   
3          27          bi analyst   
4          29        data analyst   

                                             area        city         country  \
0               ['basel', 'basel', 'switzerland']       basel     switzerland   
1          ['dublin', 'county dublin', 'ireland']      dublin         ireland   
2  ['dijon', 'bourgogne-franche-comté', 'france']       dijon          france   
3               ['alsónémedi', 'pest', 'hungary']  alsónémedi         hungary   
4     ['manchester', 'england', 'united kingdom']  manchester  united kingdom   

  workplace                              company_name  \
0   on-site  pharmiweb.jobs: global life science jobs   
1   on-site  pharmiweb.jobs: global life science jobs   
2    hybrid                                  neos-sdi   
3    hybrid                                       gls   
4   on-site                                axon moore   

                            industry company_size  \
0              staffing & recruiting        11-50   
1              staffing & recruiting        11-50   
2  information technology & services       51-200   
3   transportation/trucking/railroad      10,001+   
4              staffing & recruiting       51-200   

                                     job_description  posted_date  applicants  \
0  about the job\n        \n\n \n                ...   1 week ago          47   
1  about the job\n        \n\n \n                ...   1 week ago          30   
2  about the job\n        \n\n \nthe Dijon data t...   1 week ago         141   
3  about the job\n        \n\n \ngls' dynamically...   1 week ago          36   
4  about the job\n        \n \n\n\n\n      this j...  3 weeks ago          81   

  right_posted_date                                        hard_skills  
0        2024-01-09  [database, sql, data mining, excel, analytics,...  
1        2024-01-09                                 [machine learning]  
2        2024-01-09  [sql, power bi, business intelligence, data ma...  
3        2024-01-09          [database, sql, power bi, tableau, agile]  
4        2023-12-26                    [python, sql, excel, analytics]

### Подготовка данных для дашборда: удаление лишних столбцов

In [30]:
#Изучаем имеющиеся столбцы
filtered_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 615 entries, 0 to 614
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Unnamed: 0         615 non-null    int64 
 1   job_title          615 non-null    object
 2   area               615 non-null    string
 3   city               615 non-null    string
 4   country            615 non-null    string
 5   workplace          615 non-null    string
 6   company_name       615 non-null    string
 7   industry           615 non-null    string
 8   company_size       615 non-null    string
 9   job_description    615 non-null    object
 10  posted_date        615 non-null    string
 11  applicants         615 non-null    int32 
 12  right_posted_date  615 non-null    object
 13  hard_skills        615 non-null    object
dtypes: int32(1), int64(1), object(4), string(8)
memory usage: 65.0+ KB


In [31]:
# Столбец не нужен, показывает старые индексы
del filtered_data['Unnamed: 0']

In [32]:
# Столбец тоже не нужен, города и страны выделены отдельно
del filtered_data['area']

In [33]:
# Столбец больше неактуален - все хард скиллы вытащены
del filtered_data['job_description']

In [34]:
# Этот столбец полностью заменен на right_posted_date
del filtered_data['posted_date']

In [35]:
# Проверяем результат:
filtered_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 615 entries, 0 to 614
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   job_title          615 non-null    object
 1   city               615 non-null    string
 2   country            615 non-null    string
 3   workplace          615 non-null    string
 4   company_name       615 non-null    string
 5   industry           615 non-null    string
 6   company_size       615 non-null    string
 7   applicants         615 non-null    int32 
 8   right_posted_date  615 non-null    object
 9   hard_skills        615 non-null    object
dtypes: int32(1), object(3), string(6)
memory usage: 45.8+ KB


In [36]:
filtered_data = filtered_data.explode('hard_skills')

In [37]:
filtered_data.head()

job_title   city      country workplace  \
0  data analyst  basel  switzerland   on-site   
0  data analyst  basel  switzerland   on-site   
0  data analyst  basel  switzerland   on-site   
0  data analyst  basel  switzerland   on-site   
0  data analyst  basel  switzerland   on-site   

                               company_name               industry  \
0  pharmiweb.jobs: global life science jobs  staffing & recruiting   
0  pharmiweb.jobs: global life science jobs  staffing & recruiting   
0  pharmiweb.jobs: global life science jobs  staffing & recruiting   
0  pharmiweb.jobs: global life science jobs  staffing & recruiting   
0  pharmiweb.jobs: global life science jobs  staffing & recruiting   

  company_size  applicants right_posted_date  hard_skills  
0        11-50          47        2024-01-09     database  
0        11-50          47        2024-01-09          sql  
0        11-50          47        2024-01-09  data mining  
0        11-50          47        2024-01-09        excel  
0        11-50          47        2024-01-09    analytics

In [38]:
# Видим, что у некоторых столбцов сбился тип данных - исправляем
filtered_data['job_title'] = (filtered_data['job_title']
                 .astype('string')
                 .str.lower()
                 .str.strip()
                )

filtered_data['hard_skills'] = (filtered_data['hard_skills']
                 .astype('string')
                 .str.lower()
                 .str.strip()
                )
filtered_data['right_posted_date'] = pd.to_datetime(filtered_data['right_posted_date'])

In [39]:
filtered_data['index'] = filtered_data.index

In [40]:
# Итоговый результат
filtered_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2354 entries, 0 to 614
Data columns (total 11 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   job_title          2354 non-null   string        
 1   city               2354 non-null   string        
 2   country            2354 non-null   string        
 3   workplace          2354 non-null   string        
 4   company_name       2354 non-null   string        
 5   industry           2354 non-null   string        
 6   company_size       2354 non-null   string        
 7   applicants         2354 non-null   int32         
 8   right_posted_date  2354 non-null   datetime64[ns]
 9   hard_skills        2327 non-null   string        
 10  index              2354 non-null   int64         
dtypes: datetime64[ns](1), int32(1), int64(1), string(8)
memory usage: 211.5 KB


In [41]:
# Итоговый результат
filtered_data.head()

job_title   city      country workplace  \
0  data analyst  basel  switzerland   on-site   
0  data analyst  basel  switzerland   on-site   
0  data analyst  basel  switzerland   on-site   
0  data analyst  basel  switzerland   on-site   
0  data analyst  basel  switzerland   on-site   

                               company_name               industry  \
0  pharmiweb.jobs: global life science jobs  staffing & recruiting   
0  pharmiweb.jobs: global life science jobs  staffing & recruiting   
0  pharmiweb.jobs: global life science jobs  staffing & recruiting   
0  pharmiweb.jobs: global life science jobs  staffing & recruiting   
0  pharmiweb.jobs: global life science jobs  staffing & recruiting   

  company_size  applicants right_posted_date  hard_skills  index  
0        11-50          47        2024-01-09     database      0  
0        11-50          47        2024-01-09          sql      0  
0        11-50          47        2024-01-09  data mining      0  
0        11-50          47        2024-01-09        excel      0  
0        11-50          47        2024-01-09    analytics      0

In [42]:
filtered_data.to_csv('dashbord_data.csv')

### Приложение 1 - Ссылка на Дашборд

Дашборд доступен по ссылке: https://public.tableau.com/views/JobAnalysisforJuniorDataAnalysts/Dashboard1?:language=en-US&:display_count=n&:origin=viz_share_link